src/views/ModelPlan/TaskList/Basics/index.tsx

```diff
@@ -1,29 +1,480 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { Fragment, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link, Route, Switch, useHistory, useParams } from 'react-router-dom';
+import { useMutation, useQuery } from '@apollo/client';
+import {
+  Breadcrumb,
+  BreadcrumbBar,
+  BreadcrumbLink,
+  Button,
+  Dropdown,
+  IconArrowBack,
+  Label,
+  TextInput
+} from '@trussworks/react-uswds';
+import { Field, FieldArray, Form, Formik, FormikProps } from 'formik';
 
 import MainContent from 'components/MainContent';
-import NotFound, { NotFoundPartial } from 'views/NotFound';
+import PageHeading from 'components/PageHeading';
+import PageNumber from 'components/PageNumber';
+import Alert from 'components/shared/Alert';
+import AutoSave from 'components/shared/AutoSave';
+import CheckboxField from 'components/shared/CheckboxField';
+import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
+import FieldErrorMsg from 'components/shared/FieldErrorMsg';
+import FieldGroup from 'components/shared/FieldGroup';
+import modelCategoryEnum from 'constants/enums/modelCategory';
+import GetModelPlanQuery from 'queries/GetModelPlanQuery';
+import {
+  GetModelPlan,
+  GetModelPlanVariables
+} from 'queries/types/GetModelPlan';
+import { UpdateModelPlan as UpdateModelPlanType } from 'queries/types/UpdateModelPlan';
+import UpdateModelPlan from 'queries/UpdateModelPlan';
+import flattenErrors from 'utils/flattenErrors';
+import {
+  translateCmmiGroup,
+  translateCmsCenter,
+  translateModelCategory
+} from 'utils/modelPlan';
+import planBasicsSchema from 'validations/planBasics';
+import { NotFoundPartial } from 'views/NotFound';
+
+import Milestones from './Milestones';
+import Overview from './Overview';
+
+type PlanBasicModelPlanFormType = {
+  modelName: string;
+  modelCategory: string | null;
+  cmsCenters: string[];
+  cmmiGroup: string[];
+  cmsOther: string;
+};
+
+const BasicsContent = () => {
+  const { t } = useTranslation('basics');
+  const { t: h } = useTranslation('draftModelPlan');
+  const { modelId } = useParams<{ modelId: string }>();
+
+  const formikRef = useRef<FormikProps<PlanBasicModelPlanFormType>>(null);
+  const history = useHistory();
+  const [isCmmiGroupShown, setIsCmmiGroupShown] = useState(false);
+  const [showOther, setShowOther] = useState(false);
+
+  const { data } = useQuery<GetModelPlan, GetModelPlanVariables>(
+    GetModelPlanQuery,
+    {
+      variables: {
+        id: modelId
+      }
+    }
+  );
+
+  const { modelName, modelCategory, cmsCenters, cmmiGroups, cmsOther } =
+    data?.modelPlan || {};
+
+  const [update] = useMutation<UpdateModelPlanType>(UpdateModelPlan);
+
+  const handleFormSubmit = (
+    formikValues: PlanBasicModelPlanFormType,
+    redirect?: 'next' | 'back'
+  ) => {
+    update({
```

Comment:
Before the update method, we can now validate for modelName that gets submitted

```
if (!formikValues.modelName) {
      formikRef?.current?.setFieldError('modelName', 'Enter the Model name');
      return;
    }

This will only trigger on the Save&Exit submit handler, as the autosave will not get called, and the normal form submit uses a different validation trigger.  
```